packages/renderer/src/lib/appearance/appearance-utils.spec.ts

```diff
@@ -27,115 +25,115 @@ import { AppearanceUtil } from './appearance-util';
-    });
-    getConfigurationValueMock.mockResolvedValue(AppearanceSettings.SystemEnumValue);
+    } as unknown as MediaQueryList);
+    vi.mocked(window.getConfigurationValue).mockResolvedValue(AppearanceSettings.SystemEnumValue);
 
     const theme = await appearanceUtil.getTheme();
     expect(theme).toBe('dark');
   });
 
   test('should return light if OS is set to light and theme is set to system ', async () => {
-    (window as any).matchMedia = vi.fn().mockReturnValue({
+    vi.mocked(window.matchMedia).mockReturnValue({
       matches: false,
       addEventListener: vi.fn(),
       removeEventListener: vi.fn(),
-    });
-    getConfigurationValueMock.mockResolvedValue(AppearanceSettings.SystemEnumValue);
+    } as unknown as MediaQueryList);
+    vi.mocked(window.getConfigurationValue).mockResolvedValue(AppearanceSettings.SystemEnumValue);
 
     const theme = await appearanceUtil.getTheme();
 
     expect(theme).toBe('light');
   });
 
   test('should return dark if value is dark even if os is light', async () => {
-    (window as any).matchMedia = vi.fn().mockReturnValue({
+    vi.mocked(window.matchMedia).mockReturnValue({
       matches: false,
       addEventListener: vi.fn(),
       removeEventListener: vi.fn(),
-    });
-    getConfigurationValueMock.mockResolvedValue(AppearanceSettings.DarkEnumValue);
+    } as unknown as MediaQueryList);
+    vi.mocked(window.getConfigurationValue).mockResolvedValue(AppearanceSettings.DarkEnumValue);
 
     const theme = await appearanceUtil.getTheme();
     expect(theme).toBe('dark');
   });
 
   test('should return light if value is light even if os is dark', async () => {
-    (window as any).matchMedia = vi.fn().mockReturnValue({
+    vi.mocked(window.matchMedia).mockReturnValue({
       matches: true,
       addEventListener: vi.fn(),
       removeEventListener: vi.fn(),
-    });
-    getConfigurationValueMock.mockResolvedValue(AppearanceSettings.LightEnumValue);
+    } as unknown as MediaQueryList);
+    vi.mocked(window.getConfigurationValue).mockResolvedValue(AppearanceSettings.LightEnumValue);
 
     const theme = await appearanceUtil.getTheme();
     expect(theme).toBe('light');
   });
 
   test('should return custom value even if os is dark', async () => {
-    (window as any).matchMedia = vi.fn().mockReturnValue({
+    vi.mocked(window.matchMedia).mockReturnValue({
       matches: true,
       addEventListener: vi.fn(),
       removeEventListener: vi.fn(),
-    });
+    } as unknown as MediaQueryList);
     const customTheme = 'fooTheme';
-    getConfigurationValueMock.mockResolvedValue(customTheme);
+    vi.mocked(window.getConfigurationValue).mockResolvedValue(customTheme);
 
     const theme = await appearanceUtil.getTheme();
     expect(theme).toBe(customTheme);
   });
 
   test('should return custom value even if os is dark when matches found', async () => {
-    (window as any).matchMedia = vi.fn().mockReturnValue({
+    vi.mocked(window.matchMedia).mockReturnValue({
```

Comment:
it looks like this instruction should be in the beforeEach block as being copied in all the tests

Reply:
I put it here since we use true/false in each different test.

Reply:
vi.mocked(window.matchMedia).mockReturnValue is almost the same so you could return empty in before each and only override the value in a specific test

Reply:
we return an empty object in many cases